# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/kwin/hotfix-x32-pointer.patch
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- kwin-6.7.3/src/backends/drm/drm_commit.cpp.vanilla	2026-07-26 16:25:26.704436714 +0200
+++ kwin-6.7.3/src/backends/drm/drm_commit.cpp	2026-07-26 16:25:35.144010900 +0200
@@ -153,12 +153,15 @@ bool DrmAtomicCommit::doCommit(uint32_t
     drm_mode_atomic commitData{
         .flags = flags,
         .count_objs = uint32_t(objects.size()),
-        .objs_ptr = reinterpret_cast<uint64_t>(objects.data()),
-        .count_props_ptr = reinterpret_cast<uint64_t>(propertyCounts.data()),
-        .props_ptr = reinterpret_cast<uint64_t>(propertyIds.data()),
-        .prop_values_ptr = reinterpret_cast<uint64_t>(values.data()),
+        .objs_ptr = reinterpret_cast<uintptr_t>(objects.data()),
+        .count_props_ptr = reinterpret_cast<uintptr_t>(propertyCounts.data()),
+        .props_ptr = reinterpret_cast<uintptr_t>(propertyIds.data()),
+        .prop_values_ptr = reinterpret_cast<uintptr_t>(values.data()),
         .reserved = 0,
-        .user_data = reinterpret_cast<uint64_t>(this),
+        .user_data = reinterpret_cast<uintptr_t>(this),
     };
     return drmIoctl(m_gpu->fd(), DRM_IOCTL_MODE_ATOMIC, &commitData) == 0;
 }
